include/xtensor-signal/lfilter.hpp

```diff
@@ -0,0 +1,92 @@
+#ifndef XTENSOR_SIGNAL_LFILTER_HPP
+#define XTENSOR_SIGNAL_LFILTER_HPP
+
+#include <type_traits>
+#include <xtensor-blas/xlinalg.hpp>
+#include <xtensor/xarray.hpp>
+#include <xtensor/xaxis_slice_iterator.hpp>
+#include <xtensor/xbuilder.hpp>
+#include <xtensor/xexception.hpp>
+#include <xtensor/xindex_view.hpp>
+#include <xtensor/xnoalias.hpp>
+#include <xtensor/xpad.hpp>
+#include <xtensor/xtensor.hpp>
+
+namespace xt {
+namespace signal {
+namespace detail {
```

Comment:
Since `xtensor-signal` is C++17, you can simplify this either with:
```cpp
namespace xt::signal::detail
{
    // ...
}
```
(but then you need to reopen `xt::signal` for the public `lfiter` definition)
or with:
```cpp
namespace xt::signal
{
    namespace detail
    {
        // Private impl
    }
    // public impl
}